### PR TITLE
feat(console): onboarding flow

### LINKED ISSUE #5213: Onboarding Progress View
As a (new) IAM Admin, I want a better guidance so that I know what to do next as I setup my ZITADEL.

### Acceptance Criteria
- [ ]  I be led through the setup flow in ZITADEL, till I can login to my app the first time
- [ ]  I see which steps I have successfully done (see [Mid-through progress view](https://user-images.githubusercontent.com/48990649/215735392-2a2827ac-8d0d-47e1-b4d7-9c23176bc28b.png))
- [ ]  I see wich steps are missing and how to get to the next (see [Empty state view](https://user-images.githubusercontent.com/48990649/215735384-5cbabab5-f451-4661-a35e-d1e0a5e56de4.png))
- [ ]  I can opt-out the setup flow help (which will be saved in the user metadata)
- [ ]  For users that finished or opt-out the progress we show the standard boxes (see [Completed Progress View](https://user-images.githubusercontent.com/48990649/215735394-5d9889b7-54ba-4900-872a-db110e08e3f4.png))

**Empty state of the progress**
zitadel-onboarding02a
1. A button to initialize a wizard that guides you through each necessary steps
2. re-arranged the checkmarks and visual empty ones on non-finished steps
3. highlighting a next suggested step

**Mid-through progress view**
zitadel-onboarding02b
4. fading out, strike-through and re-arranged to the end of the list steps that have been done.

**Completed Progress View**
zitadel-onboarding02c
With this view, the cards can act as shortcuts for further navigation (and it makes it faster as well) even after the setup progress is completed.

**Additional Information**
- We can use the event api to get the information about what has already been created.

### PR DESCRIPTION
**Pull request by peintnermax**
Closes #5213

Show an overview of the setup progress of the instance, and task that still have to be done to complete

Bildschirm_foto 2023-02-21 um 16 40 52

```[tasklist]
### Definition of Ready
- [ ] I am happy with the code
- [ ] Short description of the feature/issue is added in the pr description
- [ ] PR is linked to the corresponding user story
- [ ] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [ ] No debug or dead code
- [ ] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [ ] Functionality of the acceptance criteria is checked manually on the dev system.
```

### CODE REVIEW

**Comment by hifabienne** [comment]
@peintnermax I just had a look at your picture and it looks already good. One think that came to my mind. Are you able to define the order? Because I think we want to guide them along what is essential in the first place. that would be creating project and than an application.

**Comment by peintnermax** [comment]
> @peintnermax I just had a look at your picture and it looks already good. One think that came to my mind. Are you able to define the order? Because I think we want to guide them along what is essential in the first place. that would be creating project and than an application.

Yes thought of that.. I also thought of marking some actions as necessary or optional but that would require some changes in the design too..

**Comment by adlerhurst** [comment]
smtp settings are still unresolved after i configured them

File: console/src/assets/i18n/en.json [hunk 1]
```diff
@@ -42,6 +42,41 @@
       "ADD": "Hold and drag a tile to add"
     }
   },
+  "ONBOARDING": {
```
Comment (adlerhurst) [inline console/src/assets/i18n/en.json:45]: translation missing: 
image

File: console/src/app/modules/nav/nav.component.html [hunk 2]
```diff
@@ -212,7 +212,46 @@
 
 <ng-template #shortcutKeyRef>
   <ng-container *ngIf="(isHandset$ | async) === false">
-    <span class="fill-space"></span>
+    <ng-template cnslHasRole [hasRole]="['iam.read']">
```
Comment (adlerhurst) [inline console/src/app/modules/nav/nav.component.html:215]: this box doesn't work at all
image

**Review by adlerhurst — CHANGES_REQUESTED** [review]

File: console/src/assets/i18n/en.json (hunk 1 above)
Comment (peintnermax) [reply console/src/assets/i18n/en.json:45]: this card was just to test UI. Removed it.

File: console/src/app/modules/nav/nav.component.html (hunk 2 above)
Comment (peintnermax) [reply console/src/app/modules/nav/nav.component.html:215]: removed

File: console/src/app/modules/nav/nav.component.ts [hunk 3]
```diff
@@ -1,17 +1,20 @@
 import { animate, keyframes, style, transition, trigger } from '@angular/animations';
 import { BreakpointObserver } from '@angular/cdk/layout';
+import { ConnectedPosition, ConnectionPositionPair } from '@angular/cdk/overlay';
 import { HttpClient } from '@angular/common/http';
 import { Component, ElementRef, Input, OnDestroy, ViewChild } from '@angular/core';
 import { UntypedFormControl } from '@angular/forms';
 import { Router } from '@angular/router';
-import { BehaviorSubject, combineLatest, forkJoin, map, merge, Observable, Subject, switchMap, take } from 'rxjs';
+import { BehaviorSubject, combineLatest, map, merge, Observable, Subject, switchMap, take, tap } from 'rxjs';
```
Comment (adlerhurst) [inline console/src/app/modules/nav/nav.component.ts:8]:
```suggestion
import { BehaviorSubject, combineLatest, map, Observable, Subject, take, tap } from 'rxjs';
```

File: console/src/app/modules/nav/nav.module.ts [hunk 4]
```diff
@@ -13,18 +13,23 @@ import { HasRoleModule } from 'src/app/directives/has-role/has-role.module';
 import { HasRolePipeModule } from 'src/app/pipes/has-role-pipe/has-role-pipe.module';
 
 import { NavComponent } from './nav.component';
+import { OnboardingCardComponent } from '../onboarding-card/onboarding-card.component';
```
Comment (adlerhurst) [inline console/src/app/modules/nav/nav.module.ts:16]:
```suggestion
```

File: console/src/app/modules/onboarding/onboarding.module.ts [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+import { MatLegacyButtonModule as MatButtonModule } from '@angular/material/legacy-button';
+import { MatRippleModule } from '@angular/material/core';
+import { MatIconModule } from '@angular/material/icon';
+import { MatLegacyProgressSpinnerModule as MatProgressSpinnerModule } from '@angular/material/legacy-progress-spinner';
+import { MatLegacyTooltipModule as MatTooltipModule } from '@angular/material/legacy-tooltip';
+import { TranslateModule } from '@ngx-translate/core';
+import { HasRoleModule } from 'src/app/directives/has-role/has-role.module';
```
Comment (adlerhurst) [inline console/src/app/modules/onboarding/onboarding.module.ts:9]:
```suggestion
```

File: console/src/app/modules/onboarding/onboarding.module.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+import { MatLegacyButtonModule as MatButtonModule } from '@angular/material/legacy-button';
```
Comment (adlerhurst) [inline console/src/app/modules/onboarding/onboarding.module.ts:3]:
```suggestion
```

File: console/src/app/modules/onboarding-card/onboarding-card.component.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { AdminService } from 'src/app/services/admin.service';
+import { StorageLocation, StorageService } from 'src/app/services/storage.service';
```
Comment (adlerhurst) [inline console/src/app/modules/onboarding-card/onboarding-card.component.ts:4]:
```suggestion
```

File: console/src/app/modules/onboarding-card/onboarding-card.module.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+import { MatLegacyButtonModule as MatButtonModule } from '@angular/material/legacy-button';
+import { MatRippleModule } from '@angular/material/core';
```
Comment (adlerhurst) [inline console/src/app/modules/onboarding-card/onboarding-card.module.ts:4]:
```suggestion
```

File: console/src/app/modules/onboarding-card/onboarding-card.module.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+import { MatIconModule } from '@angular/material/icon';
+import { MatLegacyProgressSpinnerModule as MatProgressSpinnerModule } from '@angular/material/legacy-progress-spinner';
+import { MatLegacyTooltipModule as MatTooltipModule } from '@angular/material/legacy-tooltip';
+import { TranslateModule } from '@ngx-translate/core';
+import { HasRoleModule } from 'src/app/directives/has-role/has-role.module';
+import { ShortcutsModule } from 'src/app/modules/shortcuts/shortcuts.module';
```
Comment (adlerhurst) [inline console/src/app/modules/onboarding-card/onboarding-card.module.ts:10]:
```suggestion
```

File: console/src/app/modules/onboarding-card/onboarding-card.module.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+import { OnboardingCardComponent } from './onboarding-card.component';
+import { RouterModule } from '@angular/router';
+import { TimestampToDatePipeModule } from 'src/app/pipes/timestamp-to-date-pipe/timestamp-to-date-pipe.module';
+import { LocalizedDatePipeModule } from 'src/app/pipes/localized-date-pipe/localized-date-pipe.module';
```
Comment (adlerhurst) [inline console/src/app/modules/onboarding-card/onboarding-card.module.ts:15]:
```suggestion
```

File: console/src/app/pages/app-create/app-create.component.ts [hunk 11]
```diff
@@ -0,0 +1,50 @@
+import { animate, style, transition, trigger } from '@angular/animations';
+import { Location } from '@angular/common';
```
Comment (adlerhurst) [inline console/src/app/pages/app-create/app-create.component.ts:2]:
```suggestion
```

File: console/src/app/pages/app-create/app-create.component.ts [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+import { Component } from '@angular/core';
+import { AbstractControl, UntypedFormBuilder, UntypedFormGroup, ValidatorFn, Validators } from '@angular/forms';
+import { MatLegacySlideToggleChange as MatSlideToggleChange } from '@angular/material/legacy-slide-toggle';
```
Comment (adlerhurst) [inline console/src/app/pages/app-create/app-create.component.ts:5]:
```suggestion
```

File: console/src/app/pages/app-create/app-create.component.ts [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+import { Router } from '@angular/router';
+import { take } from 'rxjs/operators';
```
Comment (adlerhurst) [inline console/src/app/pages/app-create/app-create.component.ts:7]:
```suggestion
```

File: console/src/app/pages/app-create/app-create.component.ts [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+import { ProjectAutocompleteType } from 'src/app/modules/search-project-autocomplete/search-project-autocomplete.component';
+import { lowerCaseValidator, numberValidator, symbolValidator, upperCaseValidator } from 'src/app/pages/validators';
+import { SetUpOrgRequest } from 'src/app/proto/generated/zitadel/admin_pb';
+import { PasswordComplexityPolicy } from 'src/app/proto/generated/zitadel/policy_pb';
+import { GrantedProject, Project } from 'src/app/proto/generated/zitadel/project_pb';
```
Comment (adlerhurst) [inline console/src/app/pages/app-create/app-create.component.ts:12]:
```suggestion
import { Project } from 'src/app/proto/generated/zitadel/project_pb';
```

File: console/src/app/pages/app-create/app-create.component.ts [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+import { Gender } from 'src/app/proto/generated/zitadel/user_pb';
+import { AdminService } from 'src/app/services/admin.service';
```
Comment (adlerhurst) [inline console/src/app/pages/app-create/app-create.component.ts:14]:
```suggestion
```

**Review by adlerhurst — COMMENTED** [review]
there are many unused imports

**Review by adlerhurst — APPROVED** [review]

**Pull request merged**